ext/mysqli/tests/gh16910.phpt

```diff
@@ -0,0 +1,40 @@
+--TEST--
+GH-xxxxx (AUTO_SEC_PART_DIGITS (field->decimal value=39) ignored in binary protocol)
+--EXTENSIONS--
+mysqli
+--SKIPIF--
+<?php
+require_once 'skipifconnectfailure.inc';
+?>
+--FILE--
+<?php
+require_once 'connect.inc';
+
+if (!$link = my_mysqli_connect($host, $user, $passwd, $db, $port, $socket)) {
+    printf("Cannot connect to the server using host=%s, user=%s, passwd=***, dbname=%s, port=%s, socket=%s\n",
+        $host, $user, $db, $port, $socket);
+    exit(1);
+}
+
+// Values from binary and text protocol should be the same
+$stmt = $link->prepare("select FROM_UNIXTIME('1992.1'), FROM_UNIXTIME('1992.0')");
+$stmt->execute();
+$stmt->bind_result($binval[0],$binval[1]);
+$stmt->fetch();
+$stmt->close();
+
+$result = $link->query("select FROM_UNIXTIME('1992.1'), FROM_UNIXTIME('1992.0')");
+$txtval = $result->fetch_row();
+$result->close();
+$err_cnt= 0;
+
+for ($i=0; $i < 2; $i++)
+    if ($txtval[$i] != $binval[$i])
+        printf("[%03d] Expected '%s' (text protocol), got '%s' (binary protocol)",
+               $err_cnt++, $txtval[$i], $binval[$i]);
+
+if ($err_cnt == 0)
+    print "done!";
+?>
+--EXPECT--
+done!
```

Comment:
So after checking the code I had a little suspicion about this test so I gave it a try and it passes without the patch so it doesn't look like it tests the actual behaviour - at least not with MySQL version (8.40). And seems like it won't probably fail in CI either. If it's DB dependent and not easy to recreate using DB in CI, then please add a direct protocol test using the [fake server](https://github.com/php/php-src/blob/master/ext/mysqli/tests/fake_server.inc) that I recently added. Also see an example for [datetime stmt (binary protocol)](https://github.com/php/php-src/blob/master/ext/mysqli/tests/ghsa-h35g-vwh6-m678-stmt-row-datetime.phpt) and [generic query usage](https://github.com/php/php-src/blob/master/ext/mysqli/tests/protocol_query_row_fetch_data.phpt). The main part is always in fake_server though.